primitives/account/src/accounts.rs

```diff
@@ -457,10 +491,23 @@ impl Accounts {
                 data: account_info.receipt,
             });
 
-            logs.append(&mut account_info.logs);
+            if inherent.ty == InherentType::Reward {
```

Comment:
It's better to comment on this behavior, to have an in-place explanation for the differentiating behavior between inherent types.